routes/offering_test.go

```diff
@@ -0,0 +1,402 @@
+//}
+
+//func (suite *AppTestSuite) TestGetOfferingsByUserID() {
+//	offeringId := Offering1.UserID
+//	t := suite.T()
+//	rr := executeReq(t, &testRouteConfig{
+//		Method:  http.MethodGet,
+//		Body:    nil,
+//		Handler: suite.app.GetOfferingsByUserID,
+//		URL:     fmt.Sprintf("/offerings/users/%s", offeringId),
+//		URLParams: map[string]string{
+//			"user_id": offeringId,
+//		},
+//	})
+//	assert.Equal(t, rr.Code, http.StatusOK, "status code")
+//	var payload []*model.Offering
+//	_ = json.Unmarshal(rr.Body.Bytes(), &payload)
+//	assert.Equal(t, 2, len(payload), "testGetOfferingsByUserID: incorrect response size")
+//	expectedOfferings := []*model.Offering{Offering1,Offering2}
+//	for _, expected := range expectedOfferings {
+//		found := false
+//		for _, o := range payload {
+//			if o.Equal(expected) {
+//				found = true
+//			}
+//		}
+//		if !found {
+//			t.Fatalf("testGetOfferingsByUserID: %s not found in offring list", expected.ID)
+//		}
+//	}
+//}
+
+//func (suite *AppTestSuite) TestGetOfferingsByDateRange() {
+//	offeringStart := "1547337600"
+//	offeringEnd := "1547510399"
+//	t := suite.T()
+//	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("/offerings?%s&%s", offeringStart, offeringEnd), nil)
+//	if err != nil {
+//		t.Fatal(err)
+//	}
+//	q := req.URL.Query()
+//	q.Add("start", offeringStart)
+//	q.Add("end", offeringEnd)
+//	req.URL.RawQuery = q.Encode()
+//	//req = mux.SetURLVars(req, config.URLParams)
+//	rr := httptest.NewRecorder()
+//	handler := http.HandlerFunc(suite.app.GetOfferingsByDateRange)
+//	handler.ServeHTTP(rr, req)
+//	assert.Equal(t, rr.Code, http.StatusOK, "status code")
+//	var payload []*model.Offering
+//	_ = json.Unmarshal(rr.Body.Bytes(), &payload)
+//	assert.Equal(t, 1, len(payload), "testGetOfferingsByDateRange: incorrect response size")
+//	expectedOfferings := []*model.Offering{Offering5}
+//	for _, expected := range expectedOfferings {
+//		found := false
+//		for _, o := range payload {
+//			if o.Equal(expected) {
+//				found = true
+//			}
+//		}
+//		if !found {
+//			t.Fatalf("testGetOfferingsByDateRange: %s not found in offring list", expected.ID)
+//		}
+//	}
+//}
+
+var startNew, _ = utils.TimeStampToTime("1583884800")
+var endNew, _ = utils.TimeStampToTime("1583971199")
+var newOffering = &model.Offering{
+	ID: "", // Unknown at this point
+	UserID: "32ea2fb1-7124-304a-b9c3-eb445578103e",
+	WorkspaceID: "5e56de3d-2323-372d-897f-23d6037c8581",
+	Cancelled: false,
+	StartDate: startNew,
+	EndDate: endNew,
+	CreatedBy: "8b5bb736-6a1d-3378-8e71-ab45fe8beb84",
+}
+
+func (suite *AppTestSuite) TestCreateOffering() {
+	// Check ID does not exist in database
+	t := suite.T()
+	rr := executeReq(t, &testRouteConfig{
+		Method:  http.MethodGet,
+		Body:    nil,
+		Handler: suite.app.GetAllOfferings,
+		URL:     "/offerings/",
+	})
+	var payload []*model.Offering
+	_ = json.Unmarshal(rr.Body.Bytes(), &payload)
+	for _, o := range payload {
```

Comment:
this should be returning a single object and not an array. Write the test as its expected rather than what makes it run. A failing test will just give us an idea of what to fix. TDD!